### ISSUE TITLE
Object colors with a plurality of parameters

### ISSUE THREAD

**Issue by jfoclpf** [visionmedia/debug#642]
When one does

`debug({a:1, b:2});`

the object properties' values are printed with colours

![image](https://user-images.githubusercontent.com/3984909/49904296-00e81b80-fe6a-11e8-9c26-da1583ae998b.png)

but when one does

`debug("string", {a:1, b:2});`

colours are gone

![image](https://user-images.githubusercontent.com/3984909/49904236-de560280-fe69-11e8-88b8-292d084c5e36.png)

just FYI

**Comment by chagris** [comment]
(just to add some context)
This is because debug automatically adds a `"%O"` string to the arguments when the first one isn't a `string`, which is then interpreted as a [formatter](https://github.com/visionmedia/debug#formatters) for the next argument; but it doesn't if the first argument is already a string. The arguments are then passed to `util.format()`, which does the pretty-print, but without the colors.

If you want to have colors when doing 
```js
debug("string", {a:1, b:2});
```
You'll need to manually add the %O:
```js
debug("string %O", {a:1, b:2});
```

**Comment by Qix-** [comment]
Yep, @chagris is exactly right. This is intended behavior.

**Comment by jfoclpf** [comment]
@Qix- 
I see the problem and thank you guys for the reply, but how come is the intended behaviour?

Why simply not

```
if (typeof args[i] === "string"){
  args[i] += " %0";
}
```

Am I missing anything here?

**Comment by Qix-** [comment]
Because this is how console.log previously worked IIRC.

I would be open to changing it.

**Comment by jfoclpf** [comment]
You can very well use the [arguments](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/arguments) array

```js
if (arguments.length > 1 && typeof arguments[0] === "string"){
  arguments[0] += " %O";
}
```

what do you think @chagris ?

**Comment by chagris** [comment]
Yeah, my guess was that it was this way for historical reasons (`util.format()` has a `%O` since v.8.4.0)

It'd be more or less easy to have it colored one way or another (debug uses the `...args` [rest parameters](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/rest_parameters)) and the [current code](
https://github.com/visionmedia/debug/blob/master/src/common.js#L85) is pretty similar to what you proposed:
```js
	args[0] = createDebug.coerce(args[0]);

	if (typeof args[0] !== 'string') {
		// Anything else let's inspect with %O
		args.unshift('%O');
	}
```
If it's not starting with a string, this part could add as many `%O` as there are arguments in one string that is unshifted, but if there's one it'd need to parse the string, count the already-existing (valid) %* placeholders and add more as needed. (also since Node v10.0 there's `util.formatWithOptions` which could automatically color everything but that'd break a lot of code that isn't running on a recent Node?)
___
However, there's currently https://github.com/visionmedia/debug/issues/582 which aims to give more control over the output format, and ultimately, if possible, to get rid of `DEBUG_COLORS` and the use of `isatty()`, which'd require changes to how the `%O` and `%o` decide to (/ are made to) color their output... If you have any idea about that, would be great to hear!
___
For now, an already supported way would be to simply [override the log method](https://github.com/visionmedia/debug#output-streams) to, for example, `console.log`
```js
const debug = require('debug');
debug.log = console.log.bind(console);
log('All those objects will be colored (if you run on a recent enough Node version)', {foo: 'bar'}, {foo: 'bar'});
```
